packages/ui-billboard/src/Billboard/props.ts

```diff
@@ -37,93 +37,94 @@ import type {
   OtherHTMLAttributes,
   PropValidators
 } from '@instructure/shared-types'
-
+import type { ViewProps } from '@instructure/ui-view'
+import { MouseEvent } from 'react'
 type BillboardOwnProps = {
-  hero?: React.ReactElement | ((...args: any[]) => any)
-  size?: 'small' | 'medium' | 'large'
-  as?: AsElementType
-  elementRef?: (element: Element | null) => void
-  heading?: string
-  headingAs?: 'h1' | 'h2' | 'h3' | 'span'
-  headingLevel?: 'h1' | 'h2' | 'h3' | 'h4'
-  message?: React.ReactNode | ((...args: any[]) => any)
-  onClick?: (...args: any[]) => any
-  href?: string
-  disabled?: boolean
-  readOnly?: boolean
-  margin?: Spacing
-}
-
-type PropKeys = keyof BillboardOwnProps
-
-type AllowedPropKeys = Readonly<Array<PropKeys>>
-
-type BillboardProps = BillboardOwnProps &
-  WithStyleProps<BillboardTheme, BillboardStyle> &
-  OtherHTMLAttributes<BillboardOwnProps>
-
-type BillboardStyle = ComponentStyle<
-  'billboard' | 'content' | 'hero' | 'heading' | 'message'
->
-
-const propTypes: PropValidators<PropKeys> = {
   /**
    * Provide an <Img> component or Instructure Icon for the hero image
    */
-  hero: PropTypes.oneOfType([PropTypes.element, PropTypes.func]),
+  hero?: React.ReactElement | ((size: 'medium' | 'x-large' | 'large') => string)
```

Comment:
The function can return `ReactElement` too. Also, I would make a type here: `type HeroIconSize = 'medium' | 'x-large' | 'large'` and use it here and and in the index too: 

```js
hero?: React.ReactElement | ((iconSize: HeroIconSize) => React.ReactElement)
```

```js
get SVGIconSize(): HeroIconSize {
    const size = this.props.size

    // serve up appropriate SVGIcon size for each Billboard size
    if (size === 'small') {
      return 'medium'
    } else if (size === 'large') {
      return 'x-large'
    } else {
      return 'large'
    }
  }
```